openarc/util.c

```diff
@@ -891,9 +892,9 @@ arcf_mkarray(char *in)
 	if (out == NULL)
 		return (const char **) NULL;
 
-	for (p = strtok(in, ","); *p != '\0'; p = strtok(NULL, ","))
-		out[c] = p;
-	out[c] = NULL;
+	for (p = strtok_r(in, ",", &ctx); p != NULL; p = strtok_r(NULL, ",", &ctx))
+		out[c++] = p;
+	out[n] = NULL;
```

Comment:
@flowerysong Nope.

The whole point of having `n` as a variable is that it's the number of entries in the array.  That's calculated in this loop:

        for (p = in; *p != '\0'; p++)
	{
		if (*p == ',')
			n++;
	}